apis/secretsmanager/v1alpha1/secret_types.go

```diff
@@ -0,0 +1,143 @@
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// Keys used in connection secret.
+const (
+	ConnectionSecretKeyName        = "secret"
+	ConnectionSecretKeyProjectName = "projectName"
+)
+
+// SecretParameters defines parameters for a desired Secret Manager's secret.
+type SecretParameters struct {
+
+	// Required. The resource name of the project to associate with the
+	// [Secret][google.cloud.secretmanager.v1.Secret], in the format `projects/*`.
+	Parent string `json:"parent,omitempty"`
+
+	// Labels are used as additional metadata on Topic.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// The replication policy of the secret data attached to the Secret. Can be automatic or userManaged.
+	// Ref: https://cloud.google.com/secret-manager/docs/reference/rest/v1/projects.secrets
+	// +optional
+	// +immutable
+	Replication *Replication `json:"replication,omitempty"`
+}
+
+// Replication policy that defines the replication configuration of data.
+type Replication struct {
+	// The replication policy for this secret.
+	// This devaites from the internal representation of the google API since it's an interface.
+	// Ref: google/cloud/secretmanager/v1/resources.proto
+	// There are two possible values : usermanaged and automatic
+	// +optional
+	// +immutable
+	ReplicationType *ReplicationType `json:"ReplicationType"`
+}
+
+// ReplicationType refers to type of replication of the secrets
+type ReplicationType struct {
+	// +optional
+	// +immutable
+	// +nullable
+	Automatic *ReplicationAutomatic `json:"ReplicationAutomatic"`
+
+	// +optional
+	// +immutable
+	UserManaged *ReplicationUserManaged `json:"ReplicationUserManaged"`
+}
+
+// ReplicationAutomatic has fields for automatic replication of secrets
+type ReplicationAutomatic struct {
+	// +optional
+	// +immutable
+	// +nullable
+}
+
+// ReplicationUserManaged has fields for user managed replication of secrets
+type ReplicationUserManaged struct {
+
+	// Locations of the secret.
+	// +immutable
+	Replicas []*ReplicationUserManagedReplica `json:"Replicas"`
+}
+
+// ReplicationUserManagedReplica has fields of each replica of user managed replica type
+type ReplicationUserManagedReplica struct {
+	// Location of the secret.
+	// +immutable
+	Location string `json:"location,omitempty"`
+}
+
+// SecretObservation is used to show the observed state of the
+// Secret resource on GCP Secrets Manager. All fields in this structure should only
+// be populated from GCP responses; any changes made to the k8s resource outside
+// of the crossplane gcp controller will be ignored and overwritten.
+type SecretObservation struct {
+	// CreateTime: Output only. The time at which this Secret was created.
+	CreateTime string `json:"createTime,omitempty"`
+
+	// Output only. This must be unique within the project. External name of the object is set to this field, hence making it optional
+	// +optional
+	SecretID *string `json:"secretid,omitempty"`
```

Comment:
```suggestion
	SecretID string `json:"secretid,omitempty"`
```
We typically do not use pointer types in status.

Reply:
I am wondering if I should remove this field as it doesn't provide much value anyway.